file.go

```diff
@@ -0,0 +1,144 @@
+package scribe
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/Hatch1fy/cron"
+	"github.com/Hatch1fy/errors"
+)
+
+const (
+	// ErrFileUnset is returned when a file is not set and attempting to be written to
+	ErrFileUnset = errors.Error("cannot write, file is currently unset")
+	// ErrRotationIntervalSet is returned when a rotation interval is attempting to be set when it's already been set
+	ErrRotationIntervalSet = errors.Error("error setting rotation interval, value already set")
+)
+
+// NewFile will return a new Scribe
+func NewFile(name, dir string) (fp *File, err error) {
+	var f File
+	f.name = name
+	f.dir = dir
+
+	if err = f.openFile(); err != nil {
+		return
+	}
```

Comment:
This pattern with two statements inline with the if is cool, will be stealing this :)

Reply:
For me, it's a really durable way to ensure I don't return my pointer unless everything occurs without error. :)